src/main/java/com/imsweb/datagenerator/naaccr/rule/patient/BirthRule.java

```diff
@@ -22,13 +26,30 @@ public BirthRule() {
     }
 
     @Override
-    public void execute(Map<String, String> record, List<Map<String, String>> otherRecords, NaaccrDataGeneratorOptions options) {
+    public void execute(Map<String, String> record, List<Map<String, String>> otherRecords, NaaccrDataGeneratorOptions options, Map<String, Object> context) {
 
         // birth date should be no later than five years prior to min dx date (or current date if min dx date not defined)
-        LocalDate maxBirthDate = options == null ? LocalDate.now().minusYears(15) : options.getMinDxDate().minusYears(5);
+        //LocalDate maxBirthDate = options == null ? LocalDate.now().minusYears(15) : options.getMinDxDate().minusYears(5);
+        LocalDate maxBirthDate = options == null ? LocalDate.now() : options.getMinDxDate().minusYears(5);
         // limit age to max 100 years
         LocalDate minBirthDate = maxBirthDate.minusYears(100);
 
+        // Based on our tumors, pick an age that's possible for these sites.
+        Integer maxAgeGroup = (Integer)context.get(CONTEXT_FLAG_MAX_AGE_GROUP);
+        if (maxAgeGroup != null)
+            if (maxAgeGroup >= 0) {
```

Comment:
These two if statement should be merged; no reason to have them one after another one like this. On a side note, if you were to keep both, the first one would need braces.